packages/x-data-grid/src/hooks/features/columns/gridColumnsUtils.ts

```diff
@@ -380,7 +380,13 @@ export const createColumnsState = ({
       }
     });
 
-    columnsState.lookup[field] = resolveProps(existingState, { ...newColumn, hasBeenResized });
+    const newColumnWithDefaultValues = resolveProps(defaultColTypeDef, newColumn);
```

Comment:
@romgrk do we really need to use `resolveProps` here to get rid of undefined values? It is breaking other things https://github.com/mui/mui-x/issues/15409 where we need to pass undefined values, and to fix https://github.com/mui/mui-x/issues/14446  I think we should handle properly undefined values for `filterOperators` before performing any operations with them?